commands/build-image.ts

```diff
@@ -72,27 +74,33 @@ export async function buildImage(dockerFileUri?: vscode.Uri): Promise<void> {
         return;
     }
 
-    const uri: Item = await resolveImageItem(folder, dockerFileUri);
-    if (!uri) { return; }
+    while (!dockerFileItem) {
+        let resolvedItem: Item | undefined = await resolveImageItem(workspaceFolder, dockerFileUri);
+        if (resolvedItem) {
+            dockerFileItem = resolvedItem;
+        } else {
+            let msg = "Couldn't find a Dockerfile in your workspace. Would you like to add Docker files to the workspace?";
+            let response = await ext.ui.showWarningMessage(msg, DialogResponses.yes, DialogResponses.no);
+            if (response === DialogResponses.no) {
+                actionContext.properties.cancelStep = msg;
```

Comment:
Users can also cancel the dialog by clicking the 'x' icon on the notification, so your 'cancelStep' telemetry here won't be entirely accurate. You could use a try/catch or you could do something like this to fix that: https://github.com/Microsoft/vscode-azureappservice/blob/e9046407094fedbb77d994c4125b2d2bfa5819b8/src/util.ts#L63

Reply:
Thanks for catching.  Changed to this for now:

            let msg = "Couldn't find a Dockerfile in your workspace. Would you like to add Docker files to the workspace?";
            actionContext.properties.cancelStep = msg;
            await ext.ui.showWarningMessage(msg, DialogResponses.yes, DialogResponses.cancel);
            actionContext.properties.cancelStep = undefined;
            await vscode.commands.executeCommand('vscode-docker.configure');
            // Try again